glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -284,46 +269,66 @@ open class GleanInternalAPI internal constructor () {
      * @param enabled When true, enable metric collection.
      */
     fun setUploadEnabled(enabled: Boolean) {
-        if (isInitialized()) {
-            val originalEnabled = getUploadEnabled()
-
-            @Suppress("EXPERIMENTAL_API_USAGE")
-            Dispatchers.API.launch {
-                LibGleanFFI.INSTANCE.glean_set_upload_enabled(enabled.toByte())
-
-                if (!enabled) {
-                    // Cancel any pending workers here so that we don't accidentally upload or
-                    // collect data after the upload has been disabled.
-                    metricsPingScheduler.cancel()
-                    // Cancel any pending workers here so that we don't accidentally upload
-                    // data after the upload has been disabled.
-                    PingUploadWorker.cancel(applicationContext)
-                }
+        // Changing upload enabled always happens asynchronous.
```

Comment:
Yes, that's the behavior I _would_ expect. From a user perspective `uploadEnabled` was set to false only _after_ the ping was submitted.

However ... (gosh this is complicated) ... because of the delay of the uploader picking it up it could still get cancelled :/